pages/swap/_swap.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <div class="h-full w-full flex flex-wrap py-4">
+    <VHocLoading :key="$route.fullPath" :status="status">
+    <div class="container">
+      <div class="mx-auto w-full sm:w-md">
+        <div class="flex flex-col mt-6">
+          <v-swap :set-market="setMarket" />
+        </div>
+      </div>
+    </div>
+    </VHocLoading>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Status, StatusType } from '@injectivelabs/utils'
+import { UiSpotMarketWithToken } from '@injectivelabs/ui-common'
+import VSwap from '~/components/partials/swap/index.vue'
+import { Modal } from '~/types'
+import { ORDERBOOK_POLLING_ENABLED } from '~/app/utils/constants'
+import { betaMarketSlugs } from '~/app/data/market'
+
+export default Vue.extend({
+  components: {
+    VSwap
+  },
+
+  data() {
+    return {
+      status: new Status(StatusType.Loading),
+      interval: 0 as any
+    }
+  },
+
+  computed: {
+    market(): UiSpotMarketWithToken | undefined {
+      return this.$accessor.spot.market
+    },
+
+    marketIsBeta(): boolean {
+      const { params } = this.$route
+
+      return betaMarketSlugs.includes(params.spot)
+    },
+
+    markets(): UiSpotMarketWithToken[] {
+      return this.$store.state.spot.markets
+    }
+  },
+
+  watch: {
+    markets: {
+      handler () {
+        const market = this.getMarketFromRoute()
+        if (market) {
+          this.setMarket(market.slug)
+        }
+      },
+      immediate: true
+    }
+  },
+
+  mounted() {
+    const market = this.getMarketFromRoute()
+    if (!market) {
+      this.$accessor.spot.getMarkets()
+        .then(() => {
+          this.status.setIdle()
+        })
+    }
+  },
+
+  beforeDestroy() {
+    this.$accessor.spot.reset()
+    this.$accessor.modal.reset()
+    clearInterval(this.interval)
+  },
+
+  methods: {
+    getMarketFromRoute(): UiSpotMarketWithToken | undefined {
+      const { from, to } = this.$route.query
+      const market = this.markets.find((m: any) => {
+        const [base, quote] = m.slug.split('-')
+        return (from === base || from === quote) && (to === base || to === quote)
+      })
+      return market
+    },
+
+    setOrderbookPolling() {
+      if (ORDERBOOK_POLLING_ENABLED) {
+        this.interval = setInterval(async () => {
+          await this.$accessor.derivatives.pollOrderbook()
+        }, 2000)
+      }
+    },
+
+    setMarket(slug: string) {
+      this.$accessor.spot.reset()
+      this.$accessor.spot
+        .initMarket(slug)
+        .then(() => {
+          this.setOrderbookPolling()
+          this.$accessor.spot.initMarketStreams()
+        })
+        .catch(this.$onRejected)
+        .finally(() => {
+          this.status.setIdle()
+
+          if (this.marketIsBeta) {
+            this.$accessor.modal.openModal(Modal.MarketBeta)
+          }
+        })
+
+      Promise.all([
+        this.$accessor.spot.fetchOrderbook(),
```

Comment:
as described here https://github.com/InjectiveLabs/injective-dex/pull/676/files#r845978739

lets add `this.$accessor.spot.init()` to the Promise.all list